src/masternodes/rpc_tokens.cpp

```diff
@@ -725,6 +749,113 @@ UniValue minttokens(const JSONRPCRequest& request) {
+                            {"context", RPCArg::Type::STR, RPCArg::Optional::OMITTED, "Additional data necessary for specific burn type"},
+                        }
+                    },
+                    {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG,
+                        "A json array of json objects. Provide it if you want to spent specific UTXOs",
+                        {
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+               },
+               RPCResult{
+                       "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+               },
+               RPCExamples{
+                       HelpExampleCli("burntokens", "'{\"amounts\":\"10@symbol\",\"from\":\"address\"}'")
+                       + HelpExampleCli("burntokens", "'{\"amounts\":\"10@symbol\",\"from\":\"address\",\"context\":\"consortium_member_address\"}'")
+                       + HelpExampleCli("burntokens", "'{\"amounts\":\"10@symbol\",\"from\":\"address\"}' '[{\"txid\":\"id\",\"vout\":0}]'")
+                       + HelpExampleRpc("burntokens", "'{\"amounts\":\"10@symbol\",\"from\":\"address\"}' '[{\"txid\":\"id\",\"vout\":0}]'")
+               },
+    }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD,
+                           "Cannot burn tokens while still in Initial Block Download");
+    }
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    CBurnTokensMessage burnedTokens;
+    UniValue metaObj = request.params[0].get_obj();
+
+    if (!metaObj["amounts"].isNull())
+        burnedTokens.amounts = DecodeAmounts(pwallet->chain(), metaObj["amounts"].getValStr(), "");
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"amounts\" must not be null");
+    if (!metaObj["from"].isNull())
+        burnedTokens.from = DecodeScript(metaObj["from"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"from\" must not be null");
+
+    burnedTokens.burnType = CBurnTokensMessage::BurnType::TokenBurn;
+    if (!metaObj["context"].isNull())
+        burnedTokens.context = DecodeScript(metaObj["context"].getValStr());
+
+    UniValue const & txInputs = request.params[2];
+
+    int targetHeight = chainHeight(*pwallet->chain().lock()) + 1;
+
+    for (auto const & kv : burnedTokens.amounts.balances) {
```

Comment:
This check is applied on consensus side via `SubMintedTokens` so this can be removed. When the TX is checked via `execTestTx` at the end of the RPC call it will be rejected if the token is not found.